src/main/java/dan200/computercraft/shared/peripheral/monitor/BlockMonitor.java

```diff
@@ -85,9 +86,12 @@ public void setPlacedBy( @Nonnull World world, @Nonnull BlockPos pos, @Nonnull B
         if( entity instanceof TileMonitor && !world.isClientSide )
         {
             TileMonitor monitor = (TileMonitor) entity;
-            monitor.contractNeighbours();
-            monitor.contract();
-            monitor.expand();
+            if (livingEntity instanceof TurtlePlayer) {
```

Comment:
```suggestion
            if (livingEntity == null || livingEntity instanceof FakePlayer) {
```

Or similar? This should hopefully capture other block placers, which will have the same issue.

It might also be worth adding a quick comment here explaining why we do this. Something along the lines of "Defer the block update if we're being placed by another TE. See #691".